share/spice/search_code/search_code.js

```diff
@@ -1,52 +1,62 @@
-function ddg_spice_search_code(api_response) {
+(function(env) {
     "use strict";
+    env.ddg_spice_search_code = function(api_result) {
 
-    var query = api_response.query;
-    var data = api_response.results;
+        var query = api_result.query;
+        var data = api_result.results;
 
-    if(!data.length || data.length === 0) {
-        return Spice.failed('search_code');
-    }
-
-    var result;
-    for (var i = 0; i < data.length; i++) {
-        var checkRelevancy = [data[i].name, data[i].displayname, data[i].namespace].join(" ");
-        if (DDG.isRelevant(checkRelevancy, [], 2)) {
-            result = data[i];
-            break;
+        if(!data.length || data.length === 0) {
+            return Spice.failed('search_code');
         }
-    }
 
-    if (!result) {
-        return Spice.failed('search_code');
-    }
+        var result;
+        for (var i = 0; i < data.length; i++) {
+            var checkRelevancy = [data[i].name, data[i].displayname, data[i].namespace].join(" ");
+            if (DDG.isRelevant(checkRelevancy, [], 2)) {
+                result = data[i];
+                break;
+            }
+        }
 
-    function formatName(result) {
-        var formatted_name = result.name;
+        if (!result) {
+            return Spice.failed('search_code');
+        }
 
-        if (result.displayname !== '' || result.namespace !== '') {
-            formatted_name += ' (';
-            if (result.displayname !== '') {
-                formatted_name += result.displayname;
-            }
-            if (result.namespace !== '') {
-                formatted_name += (result.displayname ? ', ' : '') + result.namespace;
+        function formatName(result) {
+            var formatted_name = result.name;
+
+            if ((result.displayname && result.displayname !== '') || (result.namespace && result.namespace !== '')) {
+                formatted_name += ' (';
+                if (result.displayname && result.displayname !== '') {
+                    formatted_name += result.displayname;
+                }
+                if (result.namespace && result.namespace !== '') {
+                    formatted_name += (result.displayname ? ', ' : '') + result.namespace;
+                }
+                formatted_name += ')';
             }
-            formatted_name += ')';
+
+            return formatted_name;
         }
 
-        return formatted_name;
+        result.title = formatName(result);
+
+        Spice.add({
+            id: 'search_code',
+            name: 'Answer',
+            data: result,
+            meta: {
+                sourceURL: 'https://searchcode.com/?q=' + query,
+                sourceName: 'search[code]',
+                sourceIcon: true
+            },
+            templates: {
+                group: 'base',
```

Comment:
@moollaza May be I don't understand this properly.
How will things be different if we use `text` template instead of `base`? Just that we will have `subtitle` for showing extra information. But that can be done keeping the `base` template as well.